### ISSUE TITLE
docker pull fails on 17.10.0-ce-mac36 (19824)

### ISSUE THREAD

**Issue by mangan77** [docker/for-mac#2182]
### Expected behavior
Docker should pull image from DockerHub

### Actual behavior
Pulling image fails with timeout error.

### Information
Diagnostic ID: 0C911D1A-465F-453B-B1FB-F4E5046B8F88

__docker version__
```
Client:
 Version:      17.10.0-ce
 API version:  1.33
 Go version:   go1.8.3
 Git commit:   f4ffd25
 Built:        Tue Oct 17 19:00:43 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.10.0-ce
 API version:  1.33 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   f4ffd25
 Built:        Tue Oct 17 19:05:23 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

__docker info__
```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 17.10.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 0351df1c5a66838d0c392b4ac4cf9450de844e2d
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.44-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: linuxkit-025000000001
ID: 3MRD:2CN2:M6ZB:T3DB:AY5B:AFHI:UF3F:TGCQ:LBET:WGFZ:UPCK:LCQR
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 19
 Goroutines: 32
 System Time: 2017-10-28T17:50:41.989935394Z
 EventsListeners: 2
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

__Logs (syslog -k Sender Docker) with restart and attempting to pull image:__
```
Oct 28 20:00:32 XXXXXXXXXX Docker[3164] <Notice>: DNS lookup pool.ntp.org A: pool.ntp.org <IN|79> [A (193.228.143.12)], pool.ntp.org <IN|79> [A (83.251.214.239)], pool.ntp.org <IN|79> [A (193.228.143.24)], pool.ntp.org <IN|79> [A (62.209.166.40)]
--- last message repeated 1 time ---
Oct 28 20:00:32 XXXXXXXXXX Docker[3164] <Notice>: DNS lookup pool.ntp.org AAAA: Timeout
--- last message repeated 1 time ---
Oct 28 20:00:45 XXXXXXXXXX Docker[3163] <Error>: Fatal unexpected exception: Socket.Closed
Oct 28 20:00:51 XXXXXXXXXX Docker[4880] <Notice>: Logging to Apple System Log
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Logging to Apple System Log
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Setting handler to ignore all SIGPIPE signals
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: vpnkit version %VERSION% from %VCS_COMMIT_ID%
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: System SOMAXCONN is 128
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Will use a listen backlog of 32
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: starting port forwarding server on port_control_url:fd:4 vsock_path:/Users/magnus/Library/Containers/com.docker.docker/Data/connect
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: DNS names  will map to local IP 192.168.65.1
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: 2 upstream DNS servers are configured
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: attempting to reconnect to database
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: hosts file has bindings for XXXXXXXXXX localhost broadcasthost XXXXXXXXXX localhost
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Add(2): DNS configuration changed to: use upstream DNS servers nameserver 192.168.1.1#53
	order 0
	search lcl
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: reconnected transport layer
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: updating connection limit to 2000
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: allowing binds to any IP addresses
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: updating resolvers to use upstream DNS servers search lcl
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Warning>: No secure random number generator available
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Add(3): DNS configuration changed to: use upstream DNS servers search lcl
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: DNS names docker.for.mac.localhost will map to local IP 192.168.65.1
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: 0 upstream DNS servers are configured
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: updating resolvers to use host resolver
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Warning>: No secure random number generator available
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Remove(3): DNS configuration changed to: use upstream DNS servers nameserver 192.168.1.1#53
	order 0
	search lcl
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Add(3): DNS configuration changed to: use host resolver
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: DNS names docker.for.mac.localhost will map to local IP 192.168.65.1
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Will use the host's DNS resolver
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Disabling transparent HTTP redirection
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Creating slirp server peer_ip:192.168.65.2 local_ip:192.168.65.1 domain_search:lcl mtu:1500 port_max_idle_time:300
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: updating resolvers to use host resolver
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: We do not need secure random numbers in Host mode
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Remove(3): DNS configuration changed to: use upstream DNS servers nameserver 192.168.1.1#53
	order 0
	search lcl
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Add(3): DNS configuration changed to: use host resolver
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: DNS names docker.for.mac.localhost will map to local IP 192.168.65.1
Oct 28 20:00:51 XXXXXXXXXX Docker[4881] <Notice>: Will use the host's DNS resolver
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Vmnet.Server.negotiate: received { magic = VMN3T; version = 22; commit = a12cd7250bcd8d689078e3e42ae4a7cf6a0cbaf3 }
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Vmnet.Server.negotiate: received Ethernet 9468ba25-387b-4966-ac98-f6792d2f380a
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Vmnet.Server.negotiate: sending { mtu = 1500; max_packet_size = 1550; client_macaddr = 02:50:00:00:00:01 }
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Vmnet.Server.listen: rebinding the primary listen callback
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Vmnet.Server.listen: starting event loop
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Client mac: 02:50:00:00:00:01 server mac: f6:16:36:bc:f9:c6
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: TCP/IP ready
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: stack connected
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: starting introspection server on: fd:5
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: starting diagnostics server on: fd:6
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: updating resolvers to use host resolver
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: We do not need secure random numbers in Host mode
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Remove(3): DNS configuration changed to: use upstream DNS servers nameserver 192.168.1.1#53
	order 0
	search lcl
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Add(3): DNS configuration changed to: use host resolver
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: DNS names docker.for.mac.localhost will map to local IP 192.168.65.1
Oct 28 20:00:52 XXXXXXXXXX Docker[4881] <Notice>: Will use the host's DNS resolver
Oct 28 20:01:05 XXXXXXXXXX Docker[4881] <Notice>: Using protocol TwoThousand msize 8192
Oct 28 20:01:05 XXXXXXXXXX Docker[4881] <Notice>: BOOTREQUEST from 02:50:00:00:00:01
Oct 28 20:01:05 XXXXXXXXXX Docker[4881] <Notice>: BOOTREPLY to 02:50:00:00:00:01 yiddr 192.168.65.2 siddr 192.168.65.1 dns 192.168.65.1 router 192.168.65.1 domain unknown
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /private
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /tmp
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /Volumes
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: transfused: mount /bin/fusermount -o allow_other,max_read=1048576,subtype=osxfs /Users
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: Negotiated transfuse notification channel for /private
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: Negotiated transfuse notification channel for /tmp
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: Negotiated transfuse notification channel for /Volumes
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: Negotiated transfuse notification channel for /Users
Oct 28 20:01:06 XXXXXXXXXX Docker[4880] <Notice>: sending continue to client
Oct 28 20:01:36 XXXXXXXXXX Docker[4881] <Notice>: DNS lookup pool.ntp.org A: pool.ntp.org <IN|79> [A (193.228.143.12)], pool.ntp.org <IN|79> [A (83.251.214.239)], pool.ntp.org <IN|79> [A (193.228.143.24)], pool.ntp.org <IN|79> [A (62.209.166.40)]
Oct 28 20:01:36 XXXXXXXXXX Docker[4881] <Notice>: DNS lookup pool.ntp.org AAAA: Timeout
Oct 28 20:02:36 XXXXXXXXXX Docker[4881] <Notice>: DNS lookup pool.ntp.org A: pool.ntp.org <IN|64> [A (193.228.143.24)], pool.ntp.org <IN|64> [A (193.228.143.12)], pool.ntp.org <IN|64> [A (83.251.214.239)], pool.ntp.org <IN|64> [A (62.209.166.40)]
Oct 28 20:02:36 XXXXXXXXXX Docker[4881] <Notice>: DNS lookup pool.ntp.org AAAA: Timeout
--- last message repeated 1 time ---
Oct 28 20:03:06 XXXXXXXXXX Docker[4881] <Notice>: DNS lookup pool.ntp.org A: pool.ntp.org <IN|64> [A (193.228.143.24)], pool.ntp.org <IN|64> [A (193.228.143.12)], pool.ntp.org <IN|64> [A (83.251.214.239)], pool.ntp.org <IN|64> [A (62.209.166.40)]
Oct 28 20:03:36 XXXXXXXXXX Docker[4881] <Notice>: DNS lookup pool.ntp.org A: Timeout
Oct 28 20:03:36 XXXXXXXXXX Docker[4881] <Notice>: DNS lookup pool.ntp.org A: pool.ntp.org <IN|64> [A (193.228.143.24)], pool.ntp.org <IN|64> [A (193.228.143.12)], pool.ntp.org <IN|64> [A (83.251.214.239)], pool.ntp.org <IN|64> [A (62.209.166.40)]
```

__OS:__ macOS 10.13 High Sierra

### Steps to reproduce the behavior

  1. In terminal: `docker pull ubuntu`
  2. Pull will fail with the following message:
`Using default tag: latest
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`

This have worked before without any issue. I've tried the following:
* Reinstall - didn't help
* Moving down to stable - didn't help
* Adding `{dns : [8.8.8.8, 8.8.4.4] } - didn't help

Seems like pool.ntp.org can't be reached, but a simple `ping pool.ntp.org` works perfectly fine.

**Comment by mangan77** [comment]
Anyone got any type of feedback regarding my issue?

**Comment by johnlabarge** [comment]
I'm having the same issue

**Comment by johnlabarge** [comment]
Containers: 6
 Running: 0
 Paused: 0
 Stopped: 6
Images: 45
Server Version: 17.09.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.49-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: moby
ID: XODZ:JLWH:5347:JBBR:XEOU:MMRD:4BRW:NK56:NVVO:R6NC:GB3S:VYJM
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 18
 Goroutines: 29
 System Time: 2017-11-11T23:52:42.529144952Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Comment by karenbajador** [comment]
I am having the same issue on Docker version 17.10.0-ce, build f4ffd25

**Comment by nxmohamad** [comment]
Any updates on this?

**Comment by mangan77** [comment]
As for my issue, it "resolved" it self when I did a clean install of my Mac :)

**Comment by russtoku** [comment]
I resolve the problem of "docker pull ubuntu" failing by correcting my list of DNS servers in the Network Preferences for my active connection. I had included an inaccessible public DNS server as the first in the list. After removing it and only having Google's DNS servers in the list, the pull worked.

Interestingly, I was able to do "curl https://registry-1.docker.io/v2/" before correcting my list of DNS servers. It returned a JSON document that said "UNAUTHORIZED" which is to be expected because no credentials were sent along with the request.
